WordPress/Classes/ViewRelated/Me/All Domains/Views/AllDomainsListCardView.swift

```diff
@@ -0,0 +1,148 @@
+import SwiftUI
+
+struct DomainListCard: View {
```

Comment:
This file is not entirely new. It was first implemented in this [PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/21618). 

I changed the file's name and moved it to another folder. I also made some changes to the implementation. Feel free to review the entire file and I'll address the feedback in this PR.